src/target/riscv/riscv-013.c

```diff
@@ -2811,6 +2815,162 @@ void riscv013_fill_dmi_nop_u64(struct target *target, char *buf)
 	buf_set_u64((unsigned char *)buf, DTM_DMI_ADDRESS_OFFSET, info->abits, 0);
 }
 
+static int riscv013_test_sba_config_reg(struct target *target, uint32_t address)
+{
+  LOG_INFO("Testing System Bus Access as defined by RISC-V Debug Spec v0.13");
+
+  if(!riscv_rtos_enabled(target)) {
+    LOG_ERROR("Please run with -rtos riscv to run SBA test.");
+  }
+
+  uint32_t rd_val;
+  uint32_t sbcs_orig;
+  dmi_read(target, &sbcs_orig, DMI_SBCS);
+
+  uint32_t sbcs = sbcs_orig;
+
+  // Test 1: Simple write/read test, no address autoincrement
+  sbcs = set_field(sbcs_orig,DMI_SBCS_SBAUTOINCREMENT,0);
+  dmi_write(target,DMI_SBCS,sbcs);
+
+  for(int sbaccess = 0; sbaccess < 0x5; sbaccess++) {
+    sbcs = set_field(sbcs_orig,DMI_SBCS_SBACCESS,sbaccess);
+    dmi_write(target,DMI_SBCS,sbcs);
+
+    for(int i = 0; i < 100; i++){
+      uint32_t addr = 0x80000000 + (i << sbaccess);
+      write_memory_sba_simple(target,addr,i,sbcs_orig);
+    }
+
+    for(uint32_t i = 0; i < 100; i++){
+      uint32_t addr = 0x80000000 + (i << sbaccess);
+      uint32_t val = read_memory_sba_simple(target,addr,sbcs_orig);
+      if(i != val) {
+        LOG_ERROR("System Bus Access Test: Error reading non-autoincremented address %x, expected val = %d, read val = %d",addr,i,val);
```

Comment:
If you print this error, below it will still print out that the test passed.

Reply:
Fixed